backend/app/pre_start.py

```
@@ -0,0 +1,49 @@
+    stop=stop_after_attempt(max_tries),
+    wait=wait_fixed(wait_seconds),
+    before=before_log(logger, logging.INFO),
+    after=after_log(logger, logging.WARNING),
+    retry=retry_if_exception_type(Exception)
+    & ~retry_if_exception_type(NotImplementedError),
+)
+def init(db_engine) -> None:
+    """code to do the pre-start service"""
+    if db_engine is None:
+        logger.error("Database pool is not initialized")
+        raise RuntimeError("Database pool is not initialized")
```

Ruff (0.15.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)